src/uct/tcp/tcp_ep.c

```diff
@@ -499,8 +500,9 @@ static inline unsigned uct_tcp_ep_send(uct_tcp_ep_t *ep, size_t *sent_length)
 
     status = ucs_socket_send_nb(ep->fd, UCS_PTR_BYTE_OFFSET(ep->tx.buf, ep->tx.offset),
                                 sent_length, NULL, NULL);
-    if (status != UCS_OK) {
-        return 0;
+    if (ucs_unlikely((status != UCS_OK) &&
+                     (status != UCS_ERR_NO_PROGRESS))) {
+        return status;
```

Comment:
maybe return -1 instead?
BTW, why need to return `ssize_t`

Reply:
we would like to return an error to a user for AM functions, so we need to return `ucs_status_t` rather just negative value (`-1`)